mm2src/coins/eth.rs

```diff
@@ -1336,15 +1350,32 @@ impl MarketCoinOps for EthCoin {
         let swap_contract_address = try_tx_fus!(swap_contract_address.try_to_address());
 
         let func_name = match self.coin_type {
-            EthCoinType::Eth => "ethPayment",
-            EthCoinType::Erc20 { .. } => "erc20Payment",
+            EthCoinType::Eth => {
+                if secret_hash.len() == 32 {
+                    "ethPaymentSha256"
```

Comment:
@sergeyboyko0791 It's in TODO list already 🙂 https://github.com/KomodoPlatform/atomicDEX-API/issues/1432#issuecomment-1250850653

> Refactor the secret hash handling to Enum instead of checking its length.